### DISCUSSION TITLE
Refactoring charge declaration

### DISCUSSION THREAD

**Discussion by Blipz** [Azgaar/Armoria#113] [Ideas]
To make declaration of new charges a bit easier in `src/data/dataModel.js`, I was thinking about something like this:

```javascript
chargeData = {
  ...
  newCharge: {
    colors: 3,
    sinister: true,
    reversed: true,
    patternable: false // default value, can be omitted
  },
  ...
}
```

instead of adding things at different places in the file.

**Comment by Azgaar** [comment]
Sure, it will help. I was thinking about the same, but even more interesting:

```js
chargeData = {
  ...
  charge: {
    charge: 'name';
    p: 'e';
    elements: [
      chargeObj,
      chargeObj,
      chargeObj
   ];
  },
  ...
}
```

So charge should have required params inside and also contain an array of charges, so that charges can be nested. This way we can combine charges and create new charges out of several existing ones with different params. It would make the tool more versatile.

I have started the work in branch https://github.com/Azgaar/Armoria/tree/combined-charges, but did not have time to finish.

**Comment by Blipz** [comment]
We may not be talking about the same thing. I was referring to the description of charges in `src/data/dataModel.js`, not how they are displayed on an actual coat of arms.

**Reply by Azgaar** [reply]
Yes, but the second is a continuation of the first. We can start with format change and then enhance it into a new functionality. But I am ok on format changes as for now. As you probably now, it will also require to update the API and Map Generator as they basically use the same code.

**Comment by Blipz** [comment]
Refactoring done in 3c318a57dceb850e15ffe99ef9ca9f7aa2510f08 and 884e819ec25ef8dccd54da4e086ca61830fa0d36.
